File: cs50_x/week2/lecture/scores.c
#include <cs50.h>
#include <stdio.h>

float avg(int n, int scores[]);

// Calculates avg value of scores inputed by user.
int main(void)
{
    const int n = get_int("Amount of subjects: ");

    int scores[n];

    for (int i = 0; i < n; i++)
    {
        scores[i] = get_int("score_%i = ", i + 1);
    }

    printf("%f\n", avg(n, scores));
}

// Calcuates average float value of integers of an array
float avg(int n, int scores[])
{
    int sum = 0;

    for (int i = 0; i < n; i++)
    {
        sum = scores[i] + sum;
    }

    return (sum / (float) n);
}

File: cs50_x/week6/lecture/scores.py
# Calculates the average score.

from cs50 import get_int

def main():
    # Gets user input.
    size = pos_get_int("Amount of subjects: ")
    scores = []
    for i in range(size):
        score = pos_get_int(f"score {i + 1}: ")
        scores.append(score)

    # Prints output(i.e Average score).
    print(f"Average score: {sum(scores) / len(scores)}")

# Forces user input to be a positive integer.
def pos_get_int(prompt):
    while True:
        x = get_int(prompt)
        if x > 0:
            return x

main()